app/models/mixins/inter_region_api_method_relay.rb

```diff
@@ -0,0 +1,80 @@
+module InterRegionApiMethodRelay
+  def self.extended(klass)
+    unless klass.const_defined?("InstanceMethodRelay")
+      instance_relay = klass.const_set("InstanceMethodRelay", Module.new)
+      klass.prepend(instance_relay)
+    end
+
+    unless klass.const_defined?("ClassMethodRelay")
+      class_relay = klass.const_set("ClassMethodRelay", Module.new)
+      klass.singleton_class.prepend(class_relay)
+    end
+  end
+
+  def api_relay_method(method, action = method)
+    relay = const_get("InstanceMethodRelay")
+    collection_name = collection_for_class
+
+    relay.class_eval do
+      define_method(method) do |*meth_args, &meth_block|
+        api_args = block_given? ? yield(*meth_args) : {}
+
+        if in_current_region?
+          super(*meth_args, &meth_block)
+        else
+          api_client = InterRegionApiMethodRelay.api_client_connection_for_region(region_number)
+          collection = api_client.send(collection_name)
+          obj = collection.find(id)
+          obj.public_send(action, api_args)
```

Comment:
Also, @Fryguy how does query relation return the results of a `where` in this case? Can I call the action directly on that result or will I need a `.first`?